taca/utils/undetermined.py

```diff
@@ -0,0 +1,164 @@
+import re
+import subprocess
+import gzip
+import glob
+import os
+import logging
+import taca.flowcell_parser.classes as cl
+from taca.utils.config import CONFIG
+
+logger=logging.getLogger(__name__)
+#dmux_folder=CONFIG['analysis']['bcl2fastq']['options']['output_dir']
+dmux_folder='Demultiplexing'
+
+def check_undetermined_status(run,und_tresh=10, q30_tresh=80, freq_tresh=40, status='COMPLETED'):
+    """Will check for undetermined fastq files, and perform the linking to the sample folder if the
+    quality thresholds are met.
+    """
+    if os.path.exists(os.path.join(run, dmux_folder)):
+        xtp=cl.XTenParser(run)
+        ss=xtp.samplesheet
+        lb=xtp.lanebarcodes
+        path_per_lane=get_path_per_lane(run, ss)
+        barcode_per_lane=get_barcode_per_lane(ss)
+        workable_lanes=get_workable_lanes(run, status)
+        for lane in workable_lanes:
+            if is_unpooled_lane(ss,lane):
+               if check_index_freq(run,lane, freq_tresh):
+                    if first_qc_check(lane,lb, und_tresh, q30_tresh):
+                        link_undet_to_sample(run, lane, path_per_lane)
+            else:
+                logger.warn("The lane {}  has been multiplexed, according to the samplesheet and will be skipped.".format(lane))
+
+    else:
+        logger.warn("No demultiplexing folder found, aborting")
+
+def get_workable_lanes(run, status):
+    """List the lanes that have a .fastq file"""
+    lanes=[]
+    pattern=re.compile('L00([0-9])')
```

Comment:
yes, but part I am interested with is the lane number. If someone decides to name a file L003 in the middle of the Demultiplexing step, that is a problem, yes.